app/src/screen-components/queue/list-item.tsx

```diff
@@ -0,0 +1,256 @@
+          flexDirection: "row",
+          justifyContent: "space-between",
+          paddingStart: 16,
+          paddingEnd: 16,
+          paddingBottom: 12,
+          paddingTop: 12,
+          backgroundColor: isSelected ? theme.colors.primary : "transparent",
+        }}
+      >
+        <TouchableOpacity
+          style={{ width: 56, height: 42 }}
+          onPress={onIconPress}
+        >
+          <Image
+            style={{
+              width: "100%",
+              height: "100%",
+              borderRadius: 10,
+            }}
+            source={{
+              uri: item.img,
+            }}
+          />
+          <View
+            style={{
+              backgroundColor: "black",
+              borderRadius: 15,
+              display: "flex",
+              justifyContent: "center",
+              alignItems: "center",
+              height: 30,
+              width: 30,
+              marginTop: -36,
+              marginLeft: 12,
+            }}
+          >
+            <MaterialCommunityIcons
+              name={isPlaying ? "pause" : "play"}
+              color="white"
+              size={24}
+            />
+          </View>
+        </TouchableOpacity>
+
+        <TouchableOpacity
+          onPress={onPress}
+          style={{
+            flex: 1,
+            marginStart: 16,
+            marginEnd: 8,
+          }}
+        >
+          <View style={{}}>
+            <Text
+              style={StyleSheet.compose(fonts.titleXSmall, {
+                color: theme.colors.textInverse,
+                lineHeight: 20,
+              })}
+              numberOfLines={3}
+            >
+              {item.title}
+            </Text>
+            <Text
+              style={StyleSheet.compose(fonts.textXSmall, {
+                color: theme.colors.textLight,
+              })}
+            >
+              {formatted} ‧ {item.artist}
+            </Text>
+          </View>
+        </TouchableOpacity>
+      </View>
+    </View>
+  );
+};
+
+export const IncompleteItem = ({
+  item,
+  progress,
+  isPlaying,
+  onPress,
+  onIconPress,
+  onClearIconPress,
+}) => {
+  const theme = useTheme();
+  const fonts = useFonts();
+  const [progressWidth, setProgressWidth] = useState(0);
+
+  var formatted = "-";
+  if (item.publishedAt) {
+    if (typeof item.publishedAt === "string") {
+      item.publishedAt = parseISO(item.publishedAt);
+    }
+    formatted = format(item.publishedAt, "do MMMM, HH:mm", { locale: cs });
+  }
+
+  useEffect(() => {
+    const lastPosition = item.lastPosition ?? 0;
+    const duration = item.duration ?? 1;
+    setProgressWidth((lastPosition / duration) * 100);
+  }, []);
+
+  useEffect(() => {
+    if (progress != null) {
+      const lastPosition = Math.max(progress.position, item.lastPosition ?? 0);
+      const duration = progress.duration;
+      setProgressWidth((lastPosition / duration) * 100);
+    }
+  }, [progress]);
+
+  const styles = StyleSheet.create({
```

Comment:
styles by měly být definované mimo komponentu, jinak se definují při každém renderu... theme.colors proměnné je pak potřeba mergnout v `style={[styles.card, { backgroundColor: theme.colors.cardInverted }]}`

respektive nikdy jsem to nezkoušel takhle definovat, ale myslím, že je to méně optimalizované